### PR TITLE
Return the name/key from the enum value or undefined

### PR DESCRIPTION
**Pull request by sbosell**
Addition of the getNameFromValue method to return the key from the value.  There could be cases you may have the enum value (ie, 1) and want the key name (ie, active) especially when coming from a webservice that is returning the value and you want to show the key name.

Addition of a few tests.

### CODE REVIEW

File: src/enumValues.d.ts [hunk 1]
```diff
@@ -4,5 +4,6 @@ export declare class EnumValues {
         value: number;
     }[];
     static getNames(e: any): string[];
+    static getNameFromValue(e: any, l: (number | string)): string;
```
Comment (slavik57) [inline src/enumValues.d.ts:7]: Rename the `l` to `value`

File: src/enumValues.test.ts [hunk 2]
```diff
@@ -41,6 +52,12 @@ describe('EnumValues', () => {
       expect(EnumValues.getNames(StringValuesTestEnum)).to.deep.equal(['A', 'B', 'C']);
     });
 
+    it('getNameFromValue should return the key for that value', () => {
```
Comment (slavik57) [inline src/enumValues.test.ts:55]: change the `key` to `name`

File: src/enumValues.test.ts [hunk 3]
```diff
@@ -19,6 +20,16 @@ describe('EnumValues', () => {
       expect(EnumValues.getValues(NumericValuesTestEnum)).to.deep.equal([0, 1, 2]);
     });
 
+    it('getNameFromValue should return the key for that value', () => {
```
Comment (slavik57) [inline src/enumValues.test.ts:23]: change the `key` to `name`

File: src/enumValues.ts [hunk 4]
```diff
@@ -7,6 +7,11 @@ export class EnumValues {
     return Object.keys(e).filter(key => isNaN(+key))
   }
 
+  static getNameFromValue(e:any, l:(number|string)) {
```
Comment (slavik57) [inline src/enumValues.ts:10]: rename `l` to `value`

File: src/enumValues.ts [hunk 5]
```diff
@@ -7,6 +7,11 @@ export class EnumValues {
...
+    var all= this.getNamesAndValues(e).filter(val=> { return val.value == l});
```
Comment (slavik57) [inline src/enumValues.ts:11]: change `var` to `const`
change `val=> { return val.value == l}` to `pair => pair.value === value`
also fix the spacing

File: src/enumValues.ts [hunk 6]
```diff
@@ -7,6 +7,11 @@ export class EnumValues {
...
+    return all.length==1 ? all[0].name : undefined;
```
Comment (slavik57) [inline src/enumValues.ts:12]: return null instead of undefined.

File: src/enumValues.test.ts [hunk 7]
```diff
@@ -19,6 +20,16 @@ describe('EnumValues', () => {
...
+      var expectedResult = "B";
+      
+      expect(EnumValues.getNameFromValue(NumericValuesTestEnum, 1)).equal(expectedResult);
+    });
+
+    it('getNameFromValue should return undefined', () => {
+      expect(EnumValues.getNameFromValue(NumericValuesTestEnum, 11)).equal(undefined);
```
Comment (slavik57) [inline src/enumValues.test.ts:30]: change `equal(...)` to `to.be.null`

File: src/enumValues.test.ts [hunk 8]
```diff
@@ -41,6 +52,12 @@ describe('EnumValues', () => {
       expect(EnumValues.getNames(StringValuesTestEnum)).to.deep.equal(['A', 'B', 'C']);
     });
 
+    it('getNameFromValue should return the key for that value', () => {
+      var expectedResult = "B";
+      
+      expect(EnumValues.getNameFromValue(StringValuesTestEnum, 'BValue')).equal(expectedResult);
+    });
+
```
Comment (slavik57) [inline src/enumValues.test.ts:60]: Add a test for not existing value to return null

File: src/enumValues.test.ts [hunk 9]
```diff
@@ -19,6 +20,16 @@ describe('EnumValues', () => {
       expect(EnumValues.getValues(NumericValuesTestEnum)).to.deep.equal([0, 1, 2]);
     });
 
+    it('getNameFromValue should return the key for that value', () => {
+      var expectedResult = "B";
+      
+      expect(EnumValues.getNameFromValue(NumericValuesTestEnum, 1)).equal(expectedResult);
+    });
+
+    it('getNameFromValue should return undefined', () => {
```
Comment (slavik57) [inline src/enumValues.test.ts:29]: should return `null` not `undefined`

**Review by slavik57 — CHANGES_REQUESTED** [review]
Thanks! It is actually very useful!
I just had a few little comments about naming stuff.

**Comment by slavik57** [comment]
Also can you please update the README.

**Comment by sbosell** [comment]
I will send another pull request in a few minutes.

**Comment by slavik57** [comment]
You can fix the things I told you and commit and push to the same branch, the commits will be added to the same pull request automatically.
Just reopen the pull request, commit and push...
It's midnight here so I will probably review the changes tomorrow.
Thanks again
